web-client/src/pages/admin/AdminInventoryPage.tsx

```diff
@@ -51,24 +57,39 @@ export default function AdminInventoryPage() {
     navigate(`/admin/socks/${sockId}`);
   };
 
+  const handleAddSock = (newSock: Sock) => {
+    console.log("New sock added:", newSock);
+    setModalOpen(false);
+
+  };
+
   return (
     <div className="admin-inventory-page h-full space-y-6 px-4 py-6 md:px-8">
+
       <section className="flex flex-col items-center justify-between gap-4 sm:flex-row">
-        <h1 className="text-3xl font-bold">Inventory management</h1>
+        <h1 className="text-3xl font-bold">Inventory Management</h1>
+
 
         <Button
-          className="ml-auto"
-          onClick={() => console.log("Navigate to Add New Product page")}
+          className="bg-black text-white hover:bg-gray-800 border-black flex items-center ml-auto"
+          onClick={() => setModalOpen(true)}
         >
-          <Plus className="mr-2" />
-          Add product
+          <Plus className="mr-2 text-white" />
+          Add New Product
```

Comment:
"Add New Product" is too long for mobile. Just make it `Add product` like it was before.